lib/routes/profile.js

```diff
@@ -2,35 +2,35 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+const Boom = require('hapi').error;
 const Joi = require('joi');
 
-const db = require('../db');
+const batch = require('../batch');
 
 module.exports = {
   auth: {
-    strategy: 'oauth',
-    scope: ['profile', 'profile:write']
+    strategy: 'oauth'
   },
   response: {
     schema: {
-      email: Joi.string().required(),
-      uid: Joi.string().required(),
+      email: Joi.string().allow(null),
+      uid: Joi.string().allow(null),
       avatar: Joi.string().allow(null),
       displayName: Joi.string().allow(null)
     }
   },
   handler: function email(req, reply) {
-    var creds = req.auth.credentials;
-    db.getSelectedAvatar(creds.user).then(function(avatar) {
-      return db.getDisplayName(creds.user).then(function(profile) {
-        return {
-          email: creds.email,
-          uid: creds.user,
-          avatar: avatar ? avatar.url : null,
-          displayName: profile && profile.displayName ?
-            profile.displayName : null
-        };
-      });
+    var scopes = (req.auth.credentials.scope || []).filter(function(scope) {
+      return scope === 'profile' || scope.indexOf('profile:') !== -1;
+    });
```

Comment:
nit: this could terminate iteration after finding a profile-related scope, rather than returning all of them.  Not worth it if it would ugly up the code though.